cpp/src/io/avro/avro.cpp

```diff
@@ -141,7 +143,9 @@ bool container::parse(file_metadata* md, size_t max_num_rows, size_t first_row)
                  --num_children) {
               int skip = 1;
               if (pos == i) {
-                col.parent_union_idx = md->schema[parent_idx].num_children - num_children;
+                // parent_idx will always be pointing to our immediate parent
+                // union at this point.
+                col.parent_union_idx = parent_idx;
```

Comment:
It would be great if someone could review this.  There's no way the existing logic could have worked, it always ended up with 0 as the value for `col.parent_union_idx`.

Reply:
Yeah agree there doesn't seem to be any code that depends on it.  I was tempted to remove it, but honestly, that entire while loop is... gnarly, to say the least.  And, yeah, I think it only applies to complex/logical types (assuming there was an actual down-stream consumer of this functionality).